cmd/utils/flags.go

```diff
@@ -2260,9 +2260,10 @@ func ParseStateScheme(ctx *cli.Context, disk ethdb.Database) (string, error) {
 }
 
 // MakeTrieDatabase constructs a trie database based on the configured scheme.
-func MakeTrieDatabase(ctx *cli.Context, disk ethdb.Database, preimage bool, readOnly bool) *trie.Database {
+func MakeTrieDatabase(ctx *cli.Context, disk ethdb.Database, preimage bool, readOnly bool, isVerkle bool) *trie.Database {
```

Comment:
decided to move the decision of whether a DB is verkle to the triedb. I am still not convinced this is the right choice, as ideally the triedb would focus on _how_ to store a trie (i.e. path-based or hash-based), not _what kind_ of trie it is (verkle or mpt).

Another reason for not putting it in the triedb is the transition: this is a decision that is normally (i.e. except in the case of a verkle-at-genesis testnet) taken long after the `triedb` has been created. So in the general case, it's not obvious to me how we "upgrade" the triedb from MPT to verkle.

@rjl493456442 any opinion here?